src/framed_read.rs

```diff
@@ -219,6 +219,15 @@ pub fn framed_read2<T>(inner: T) -> FramedRead2<T> {
     }
 }
 
+pub fn framed_read2_with_buffer<T>(inner: T, buf: BytesMut) -> FramedRead2<T> {
+    FramedRead2 {
+        inner: inner,
+        eof: false,
+        is_readable: false,
```

Comment:
@alexcrichton @mredlek In my testing https://github.com/illegalprime/tokio-io I found that `is_readable` had to be set to `true` in some cases (it should anyway since there is presumably already data in the buffer). Maybe add a check to see if the buffer is empty? Maybe also add a check to see if the buffer is at least `INITIAL_CAPACITY` and grow it if it isn't?